client/info_test.go

```diff
@@ -75,3 +75,65 @@ func TestInfo(t *testing.T) {
 		t.Fatalf("expected 3 containers, got %d", info.Containers)
 	}
 }
+
+func TestInfoWithDiscoveredDevices(t *testing.T) {
+	expectedURL := "/info"
+	client := &Client{
+		client: newMockClient(func(req *http.Request) (*http.Response, error) {
+			if !strings.HasPrefix(req.URL.Path, expectedURL) {
+				return nil, fmt.Errorf("Expected URL '%s', got '%s'", expectedURL, req.URL)
+			}
+			info := &system.Info{
+				ID:         "daemonID",
+				Containers: 3,
+				DiscoveredDevices: []system.DeviceInfo{
+					{
+						Source: "cdi",
+						ID:     "vendor.com/gpu=0",
+					},
+					{
+						Source: "cdi",
+						ID:     "vendor.com/gpu=1",
+					},
+				},
+			}
+			b, err := json.Marshal(info)
+			if err != nil {
+				return nil, err
+			}
+
+			return &http.Response{
+				StatusCode: http.StatusOK,
+				Body:       io.NopCloser(bytes.NewReader(b)),
+			}, nil
+		}),
+	}
+
+	info, err := client.Info(context.Background())
+	if err != nil {
```

Comment:
Looks like we already use gotest.tools in this file, so might as well use `assert` for this test.

Reply:
Ah, good catch! I only looked at the last test